### ISSUE TITLE
FR: Implement LoginInteractive via localapi

### ISSUE THREAD

**Issue by jtcressy** [tailscale/tailscale#4738]
### What are you trying to do?

To do just the opposite of /localapi/v0/logout, I would like to instead kick-off interactive login; basically call the function StartLoginInteractive() from LocalBackend in ipn/ipnlocal/local.go#L1686 except via the localapi

This way I can start the interactive login process (basically populate an AuthURL in backend status) from an external tool without wrapping a shell around the tailscale binary. The tailscale binary is capable of it by sending requests directly through some kind of socket which I don't yet understand. It skips the localapi http interface somehow and somewhat reminds me of gRPC instead.

### How should we solve this?

Just like the logout handler here https://github.com/tailscale/tailscale/blob/fc5839864bfbd5bfed512cbb6f05e32894e93938/ipn/localapi/localapi.go#L346
we should create a new handler called `serveInteractiveLogin` or similar that instead of calling `LocalBackend`'s `LogoutSync()` function:
https://github.com/tailscale/tailscale/blob/fc5839864bfbd5bfed512cbb6f05e32894e93938/ipn/localapi/localapi.go#L355
^ called there, defined here:
https://github.com/tailscale/tailscale/blob/fc5839864bfbd5bfed512cbb6f05e32894e93938/ipn/ipnlocal/local.go#L2830

We could simply call `StartLoginInteractive()` as defined here:
https://github.com/tailscale/tailscale/blob/fc5839864bfbd5bfed512cbb6f05e32894e93938/ipn/ipnlocal/local.go#L1686

### What is the impact of not solving this?

The only way to kick off interactive login is to run `tailscale up` with no arguments (or with `--reset`) from a shell and it would be much cleaner to use the local client (`import tailscale.com/client/tailscale`) from another golang program to execute commands to the local API

**Comment by DentonGentry** [comment]
The currently supported way to do this is `tailscale up --json`
I understand that the bug states this isn't a desired solution, nonetheless it is available now.

**Comment by jtcressy** [comment]
> The currently supported way to do this is `tailscale up --json` I understand that the bug states this isn't a desired solution, nonetheless it is available now.

I did try using this, but one annoyance is this command still hangs (blocks) until authentication completes. I worked around it by placing it in a goroutine in my application so that it runs parallel, but ideally it should eventually be an api call with context.Background().

More work is in progress, going to take a swing at updating client/tailscale next.
